### DISCUSSION TITLE
how to make parameter has same variable.parameter highlight

### DISCUSSION THREAD

**Discussion by unisgn** [helix-editor/helix#11331] [Q&A] [answered]
in some lang like rust, python
both definition and reference of parameter has the same variable.parameter highlight.

but in other lang like clang, zig
only definition of parameter highlight as variable.parameter,
reference of parameter highlight as variable.

**Comment by the-mikedavis** [comment, marked as answer]
Parameter highlights extending to references of the parameters is possible because of [locals queries](https://tree-sitter.github.io/tree-sitter/syntax-highlighting#local-variables), for example for Rust: https://github.com/helix-editor/helix/blob/master/runtime/queries/rust/locals.scm

To support that kind of highlighting for more languages you can try writing the locals queries for those languages (in `runtime/queries/zig/locals.scm` for example)
